Bearded.Utilities/Monads/Result.cs

```diff
@@ -0,0 +1,128 @@
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+
+namespace Bearded.Utilities.Monads
+{
+    public readonly struct Result<TResult, TError> : IEquatable<Result<TResult, TError>>
```

Comment:
What is your reason for making TError generic instead of Exception?

Reply:
> imagine this error type being an HTTP error type

Use an http exception with a status field? But sure, I'm fine leaving this as is. I just note that it is not much more than an Either type with fancy naming then. :)

Reply:
Sure, but then `Result` still needs to be generic in the second parameter, correct? So is the suggestion that we add the `where TError : Exception` type requirement, or am I misinterpreting?

But yes, you're right, it's basically an `Either` with different semantics.

Reply:
No concrete suggestion is intended, but yes, that generic constraint would be what it would end up as. I'm happy leaving this as is.